controllers/mysqlcluster_controller.go

```diff
@@ -1127,21 +1127,27 @@ func (r *MySQLClusterReconciler) reconcileV1RestoreJob(ctx context.Context, req
 	return nil
 }
 
-func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, req ctrl.Request, cluster *mocov1beta1.MySQLCluster) error {
+func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, cluster *mocov1beta1.MySQLCluster) error {
 	secretName := cluster.ControllerSecretName()
 	secret := &corev1.Secret{}
-	secret.SetNamespace(r.SystemNamespace)
-	secret.SetName(secretName)
-	if err := r.Delete(ctx, secret); err != nil {
-		return fmt.Errorf("failed to delete controller secret %s: %w", secretName, err)
+	err := r.Get(ctx, client.ObjectKey{Namespace: r.SystemNamespace, Name: secretName}, secret)
+	if err == nil {
+		if err := r.Delete(ctx, secret); err != nil {
```

Comment:
Since we should check the error it is a not found error or not anyway,
why don't you just Delete w/o Get?